Sources/TestingMacros/Support/DiagnosticMessage.swift

```diff
@@ -438,32 +438,106 @@ struct DiagnosticMessage: SwiftDiagnostics.DiagnosticMessage {
       )
-    default:
-      return Self(
-        syntax: Syntax(functionDecl),
-        message: "Attribute \(_macroName(attribute)) must specify \(expectedArgumentCount) arguments when used with '\(functionDecl.completeName)'",
-        severity: .error
+    }
+  }
+
+  /// Create fix-its for a diagnostic stating that the given attribute must
+  /// specify arguments since it is applied to a function which has parameters.
+  ///
+  /// - Parameters:
+  ///   - attribute: The `@Test` attribute.
+  ///   - parameters: The parameter list of the function `attribute` is applied
+  ///     to.
+  ///
+  /// - Returns: An array of fix-its to include in a diagnostic.
+  private static func _addArgumentsFixIts(for attribute: AttributeSyntax, given parameters: FunctionParameterListSyntax) -> [FixIt] {
+    let baseArguments: LabeledExprListSyntax
+    if let existingArguments = attribute.arguments {
+      guard case var .argumentList(existingLabeledArguments) = existingArguments else {
+        // If there are existing arguments but they are of an unexpected type,
+        // don't attempt to provide any fix-its.
+        return []
+      }
+
+      // If the existing argument list is non-empty, ensure the last argument
+      // has a trailing comma and space.
+      if !existingLabeledArguments.isEmpty {
+        let lastIndex = existingLabeledArguments.index(before: existingLabeledArguments.endIndex)
+        existingLabeledArguments[lastIndex].trailingComma = .commaToken(trailingTrivia: .space)
+      }
+
+      baseArguments = existingLabeledArguments
+    } else {
+      baseArguments = .init()
+    }
+
+    var fixIts: [FixIt] = []
+    func addFixIt(_ message: String, appendingArguments arguments: some Collection<LabeledExprSyntax>) {
+      var newAttribute = attribute
+      newAttribute.attributeName = newAttribute.attributeName.trimmed
+      newAttribute.leftParen = .leftParenToken()
+      newAttribute.arguments = .argumentList(baseArguments + arguments)
+      newAttribute.rightParen = .rightParenToken()
+
+      fixIts.append(FixIt(
+        message: MacroExpansionFixItMessage(message),
+        changes: [.replace(oldNode: Syntax(attribute), newNode: Syntax(newAttribute))]
+      ))
+    }
+
+    // Fix-It to add 'arguments:' with one collection. If the function has 2 or
+    // more parameters, the elements of the placeholder collection are of tuple
+    // type.
+    do {
+      let argumentsCollectionType = if parameters.count == 1, let parameter = parameters.first {
+        "[\(parameter.baseTypeName)]"
+      } else {
+        "[(\(parameters.map(\.baseTypeName).joined(separator: ", ")))]"
+      }
+
+      addFixIt(
+        parameters.count == 1 ? "Add 'arguments:'" : "Add 'arguments:' with one collection",
+        appendingArguments: [LabeledExprSyntax(label: "arguments", expression: EditorPlaceholderExprSyntax(argumentsCollectionType, type: argumentsCollectionType))]
```

Comment:
I'm not certain this would be the right placeholder text. It's, AFAIK, typically some short human-readable string. In this case, maybe "arguments" or "collection". I wouldn't expect to see the name of the type (`"[T]"`) here, and an arbitrary string in the position runs the risk of breaking Fix-It display if it happens to contain special characters.

Reply:
I see quite a few examples in the Swift compiler repo which use this pattern, where the placeholder string itself provides a hint about the _type_ of the expression which the user should replace it with. In our case here, I think it's especially helpful when there are two collection arguments because it clarifies which one is which.